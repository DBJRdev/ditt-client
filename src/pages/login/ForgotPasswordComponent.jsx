import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import { withNamespaces } from 'react-i18next';
import { ForgotPassword } from 'react-ui';
import {
  RESET_PASSWORD_FAILURE,
  RESET_PASSWORD_SUCCESS,
} from '../../resources/auth/actionTypes';
import routes from '../../routes';
import styles from './Login.scss';
import logoImage from './images/logo.svg';

class ForgotPasswordComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: null,
      error: null,
      isSubmitted: false,
    };

    this.onChangeHandler = this.onChangeHandler.bind(this);
    this.resetPasswordHandler = this.resetPasswordHandler.bind(this);
  }

  onChangeHandler(field, value) {
    this.setState({
      [field]: value,
    });
  }

  resetPasswordHandler() {
    this.props.resetPassword({
      email: this.state.email,
    }).then((response) => {
      if (response.type === RESET_PASSWORD_SUCCESS) {
        this.setState({
          error: null,
          isSubmitted: true,
        });
      } else if (response.type === RESET_PASSWORD_FAILURE) {
        this.setState({ error: response.payload.response.detail });
      }
    });

    return false;
  }

  render() {
    const { t } = this.props;
    const layout = children => (
      <div className={styles.container}>
        <img
          src={logoImage}
          width={240}
          height={141}
          className={styles.logo}
          alt={t('layout:title')}
        />
        {children}
      </div>
    );

    if (this.props.isPosting) {
      return layout(t('general:text.loading'));
    }

    if (this.state.isSubmitted) {
      return layout((
        <p className={styles.message}>
          {t('login:text.passwordReset')}
        </p>
      ));
    }

    return layout((
      <ForgotPassword
        error={this.state.error}
        footer={
          // eslint-disable-next-line jsx-a11y/anchor-is-valid
          <Link to={routes.login}>
            {t('login:action.login')}
          </Link>
        }
        submitHandler={this.resetPasswordHandler}
        onChangeHandler={this.onChangeHandler}
        title={t('layout:title')}
        translations={{
          email: t('user:element.email'),
          resetPassword: t('login:action.resetPassword'),
        }}
        usernameType="email"
      />
    ));
  }
}

ForgotPasswordComponent.propTypes = {
  isPosting: PropTypes.bool.isRequired,
  resetPassword: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

export default withNamespaces()(ForgotPasswordComponent);
