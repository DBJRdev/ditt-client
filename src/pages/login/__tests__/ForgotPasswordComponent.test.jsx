import React from 'react';
import { shallow } from 'enzyme';
import ForgotPasswordComponent from '../ForgotPasswordComponent';

describe('rendering', () => {
  it('renders correctly', () => {
    const props = {
      isPosting: false,
      isPostingFailure: false,
      resetPassword: () => {},
    };
    const tree = shallow(<ForgotPasswordComponent {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
