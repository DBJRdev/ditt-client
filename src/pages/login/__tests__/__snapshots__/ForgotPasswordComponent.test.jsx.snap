// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="container"
>
  <svgr-mock
    alt="layout:title"
    className="logo"
    height={141}
    width={302}
  />
  <div
    className="box"
  >
    <div
      className="title"
    >
      layout:title
    </div>
    <div
      className="Card__root__v9QuowQ2 Card__rootColorLight__5YoQ8fwa Card__rootRaised__pWtZp1IK"
    >
      <div
        className="Card__body__NV5VWdZw"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="FormLayout__root__VuhYS7MF FormLayout__rootFieldLayoutVertical__P52sT0c-"
          >
            <label
              className="TextField__root__-CnhpfBr TextField__isRootFullWidth__2QI4JCSw TextField__isRootInFormLayout__54LhSnIJ TextField__rootLayoutVertical__da8tPfcc TextField__isRootRequired__e3NIJBxc TextField__rootSizeMedium__LvbPjcNJ TextField__rootVariantOutline__SsMHy1HK"
              htmlFor="email"
              id="email__label"
            >
              <div
                className="TextField__label__-nAzBSUI"
                id="email__labelText"
              >
                user:element.email
              </div>
              <div
                className="TextField__field__8FOvcKqB"
              >
                <div
                  className="TextField__inputContainer__t-42E39c"
                >
                  <input
                    autoComplete="username"
                    className="TextField__input__vrAdszzS"
                    disabled={false}
                    id="email"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </label>
            <button
              className="Button__root__DtkOcjth Button__rootPriorityFilled__2C6g-0nB Button__rootColorPrimary__uQtzm8My Button__rootSizeMedium__0RytHphN Button__rootBlock__Uq7W8NY0"
              disabled={false}
              id="resetPasswordButton"
              type="submit"
            >
              <span
                className="Button__label__kD4bhIKy"
                id="resetPasswordButton__labelText"
              >
                login:action.resetPassword
              </span>
            </button>
          </div>
        </form>
        <div
          className="footer"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            login:action.login
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
